WordPress/gutenberg
# SliderControl: Create new control using imported G2 Slider component

Related:
- https://github.com/WordPress/gutenberg/issues/40507

## What?

Creates a new `SliderControl` component representing only the slider part of the `RangeControl` i.e. no number input field, no icons etc.

## Why?

This is a step toward better, more modular slider-related controls.

## How?

- Uses imported [G2 Slider component](https://github.com/ItsJonQ/g2/tree/main/packages/components/src/Slider) internally (this slider input is not exported)
- Imports `interpolate` utils from G2
- Follows current format recommended by guidelines e.g. TypeScript, Emotion, hooks/components etc.
- Leverages `BaseControl` for labelling purposes etc.
- `SliderControl` has a default `size` of 36px with an `__unstable-large` variant for the future 40px default.
- Adds customization of the thumb, track foreground and track background colors.

## Todo

- [ ] Add a component readme
- [ ] Fix mouse and hover event handling
- [ ] Fix custom tooltip positioning while at extremes of slider

## Testing Instructions

TBA...

- Fire up the Storybook and ensure the SliderControl functions appropriately
- Compare the SliderControl stories with those for the RangeControl
- Run unit tests for the Gutenberg Slider `npm run test-unit packages/components/src/slider/test`

## Screenshots or screencast 



https://user-images.githubusercontent.com/60436221/182984757-5cb84071-5bfd-41f4-9f03-92d70b9d10ac.mp4
